assets/pages/monthly-contributions/reducers/reducers.js

```diff
@@ -18,54 +25,59 @@ import type { Action } from '../actions/monthlyContributionsActions';
 
 export type State = {
   amount: number,
-  country: string,
+  currency: Currency,
+  country: IsoCountry,
   error: ?string,
   payPalType: PayPalButtonType,
 };
 
-
-// ----- Setup ----- //
-
-const initialState: State = {
-  amount: 5,
-  country: 'GB',
-  error: null,
-  payPalType: 'NotSet',
+export type CombinedState = {
+  monthlyContrib: State,
+  intCmp: string,
+  user: UserState,
+  stripeCheckout: StripeCheckoutState,
+  payPalExpressCheckout: PayPalExpressCheckoutState,
+  csrf: CsrfState,
 };
 
-
 // ----- Reducers ----- //
 
-function monthlyContrib(
-  state: State = initialState,
-  action: Action): State {
-
-  switch (action.type) {
+function monthlyContrib(amount: number, currency: Currency, country: IsoCountry) {
```

Comment:
Here you are converting **reducer** to a **higher-order reducer**. In order to increase the readability of the code, I would rename this function to be `createMonthlyContribReducer`. By convention, the **reducer** functions must have the following signature: `(state, action) -> newState` and the **higher-order reducer** is a function that takes a reducer function as an argument, and/or returns a new reducer function as a result. 
Additionally, I would go one step further and rename the file to` monthlyContributionReducer.js` but maybe is a task for another PR.
Link to doc [here](http://redux.js.org/docs/recipes/reducers/SplittingReducerLogic.html)